gitlab/resource_gitlab_environment_test.go

```diff
@@ -0,0 +1,154 @@
+package gitlab
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabEnvironment_basic(t *testing.T) {
+	var environment gitlab.Environment
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabEnvironmentConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabEnvironmentExists("gitlab_environment.environment", &environment),
+					testAccCheckGitlabEnvironmentAttributes(&environment, &testAccGitlabEnvironmentAttributes{
+						Name:        "meow",
+						ExternalURL: "https://google.com",
+					}),
+				),
+			},
+			{
+				Config: testAccGitlabEnvironmentConfigNameOnly(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabEnvironmentExists("gitlab_environment.environment", &environment),
+					testAccCheckGitlabEnvironmentAttributes(&environment, &testAccGitlabEnvironmentAttributes{
+						Name: "meow",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabEnvironmentExists(n string, environment *gitlab.Environment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		environmentID := rs.Primary.ID
+		repoName := rs.Primary.Attributes["project"]
+		if repoName == "" {
+			return fmt.Errorf("No project ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		environments, _, err := conn.Environments.ListEnvironments(repoName, nil)
+		if err != nil {
+			return err
+		}
+
+		for _, gotEnvironment := range environments {
+			if strconv.Itoa(gotEnvironment.ID) == environmentID {
+				*environment = *gotEnvironment
+				return nil
+			}
+		}
+		return fmt.Errorf("Environment does not exist")
+	}
+}
+
+type testAccGitlabEnvironmentAttributes struct {
+	Name        string
+	ExternalURL string
+}
+
+func testAccCheckGitlabEnvironmentAttributes(environment *gitlab.Environment, want *testAccGitlabEnvironmentAttributes) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if environment.Name != want.Name {
+			return fmt.Errorf("got name %q; want %q", environment.Name, want.Name)
+		}
+
+		if environment.ExternalURL != want.ExternalURL {
+			return fmt.Errorf("got external url %q; want %q", environment.ExternalURL, want.ExternalURL)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckGitlabEnvironmentDestroy(s *terraform.State) error {
```

Comment:
Ideally your destroy test would be testing the behavior of your destroy function, i.e. that the environment was deleted but the project was not deleted.

Leaving this here as a mental note for #205 and for your own edification, but don't feel obligated to do this now.